sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -461,7 +461,9 @@ public SpanContext getContext() {
 
   @Override
   public boolean isRecording() {
-    return true;
+    synchronized (lock) {
+      return !hasEnded;
```

Comment:
What do you think about switching this to volatile? Sort of nice that something seemingly trivial like checking recording state doesn't have to lock.

Reply:
I'm easy either way. I think it's worth looking into the entire thread-safety management of this class, but let's save that for another PR?